common/sgx/collateral.h

```diff
@@ -65,10 +68,47 @@ oe_result_t oe_get_sgx_quote_verification_collateral(
  * oe_get_sgx_quote_verification_collateral_from_certs().
  *
  * @param[in] args The quote verification collateral.
+ * @deprecated
  */
 void oe_free_sgx_quote_verification_collateral_args(
     oe_get_sgx_quote_verification_collateral_args_t* args);
 
+/**
+ * Fetch quote verification collateral from the quote provider given the PCK
+ * certificate, CA certificate and custom parameters for third party collateral
+ * caching service.
+ *
+ * Caller is responsbile for freeing the quote verification collateral resources
+ * by calling oe_free_sgx_quote_verification_collateral_with_params_args().
+ *
+ * @param[in] leaf_cert The PCK certificate.
+ * @param[out] args Quote verification collateral.
+ */
+oe_result_t oe_get_sgx_quote_verification_collateral_with_params_from_certs(
```

Comment:
FYI, "params" in the name violates [OE Coding Conventions](https://github.com/openenclave/openenclave/blob/master/docs/Community/DevelopmentGuide.md) which says:

> DON'T use abbreviations unless they are already well-known terms known by users (e.g., "app", "info"), or are already required for use by developers (e.g., "min", "max", "args"). Examples of bad use would be num_widgets instead of widget_count, and opt_widgets instead of option_widgets or optional_widgets.

Unlike "args", "params" isn't something that's defined by the C language/headers and so does not fall within the exception clause.   I do see a bunch of other violations in public OE headers though so at least OE is somewhat consistent in not following its own coding conventions :(

Reply:
This is a very good catch, thanks! it reminds me that my mindset was sorted of limited to the name that QPL provides to OE SDK, we actually call this as baseline, now I have a better name for all of these, params sounds odd to me now